knowledge-mcp/app/main.py

```diff
@@ -0,0 +1,328 @@
+import asyncio
+from datetime import datetime
+import app.vector_db as db
+import app.db_schema as schemas
+import app.text_processing as text_proc
+import app.embeddings as embeddings
+
+from fastmcp import FastMCP
+from pydantic import Field
+from typing import Annotated
+from fastmcp.exceptions import ToolError
+from uuid import uuid4
+import hashlib
+import base64
+from fastmcp.server.dependencies import get_http_headers
+from app.config import PORT, MCP_PATH
+
+mcp = FastMCP(
+    name="KnowledgeMCPServer",
+    on_duplicate_tools="error",
+    on_duplicate_resources="warn",
+    on_duplicate_prompts="replace",
+)
+
+
+def _get_user_id() -> str:
+    headers = get_http_headers()
+    user_id = headers.get("x-forwarded-user", None)
+    if not user_id:
```

Comment:
I think you can probably just use a default user id like `anonymous` instead of erroring out so that it can be used without proxy.

Reply:
the mcp server is designed to be bundled with the oauth proxy, so not sure it makes sense to have a default user id. but I could add a stdio mode that works without the proxy.

Reply:
ok